File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/HospitalsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DepartmentsListFormatter } from "../../Tanimlamalar/Hospitals/DepartmentsListFormatter";
import { HospitalsRow } from "./HospitalsRow";

export interface HospitalsColumns {
    HospitalId: Column<HospitalsRow>;
    Name: Column<HospitalsRow>;
    CityName: Column<HospitalsRow>;
    DistrictName: Column<HospitalsRow>;
    DepartmentList: Column<HospitalsRow>;
}

export class HospitalsColumns extends ColumnsBase<HospitalsRow> {
    static readonly columnsKey = 'Tanimlamalar.Hospitals';
    static readonly Fields = fieldsProxy<HospitalsColumns>();
}

[DepartmentsListFormatter]; // referenced types

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/Hospitals/HospitalsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;
using Tanimlamalar.Hospitals;

namespace MuayeneYonetimPortali.Tanimlamalar.Columns;

[ColumnsScript("Tanimlamalar.Hospitals")]
[BasedOnRow(typeof(HospitalsRow), CheckNames = true)]
public class HospitalsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId")]
    public int HospitalId { get; set; }
    [EditLink]
    public string Name { get; set; }
    [QuickFilter]
    public string CityName{ get; set; }
    public string DistrictName { get; set; }
    // public string DepartmentName { get; set; }
    [Width(200)][DepartmentsListFormatter]
    public List<int> DepartmentList { get; set; }

}